--- git/main/Makefile.am.orig	2015-07-31 19:14:13.421319916 +0200
+++ git/main/Makefile.am	2015-07-31 19:14:35.157317091 +0200
@@ -9,8 +9,6 @@
 
 bin_PROGRAMS = enigma2
 
-installdir = $(targetprefix)/usr/local/bin
-
 enigma2_SOURCES = \
 	bsod.cpp \
 	bsod.h \
--- git/lib/gdi/picload.cpp.orig	2015-08-01 01:10:55.666375534 +0200
+++ git/lib/gdi/picload.cpp	2015-08-01 01:15:00.510518640 +0200
@@ -477,7 +477,7 @@
 	int cmaps;
 	int extcode;
 
-	gft = DGifOpenFileName(filepara->file, &ErrorCode);
+	gft = DGifOpenFileName(filepara->file);
 	if (gft == NULL) {
 		eDebug("[Picload] Error open gif %i", ErrorCode);
 		return;
@@ -569,11 +569,11 @@
 	}
 	while (rt != TERMINATE_RECORD_TYPE);
 
-	DGifCloseFile(gft, &ErrorCode);
+	DGifCloseFile(gft);
 	return;
 ERROR_R:
 	eDebug("[ePicLoad] <Error gif>");
-	DGifCloseFile(gft, &ErrorCode);
+	DGifCloseFile(gft);
 }
 
 //---------------------------------------------------------------------------------------------
--- git/data/fonts/Makefile.am.orig	2015-08-02 00:01:42.485605191 +0200
+++ git/data/fonts/Makefile.am	2015-08-01 13:04:20.528476336 +0200
@@ -5,7 +5,11 @@
 installdir = $(datadir)/fonts
 
 dist_install_DATA = \
+	ae_AlMateen.ttf \
 	andale.ttf \
+	lcd.ttf \
+	md_khmurabi_10.ttf \
+	nmsbd.ttf \
 	LiberationSans-Bold.ttf \
 	LiberationSans-BoldItalic.ttf \
 	LiberationSans-Italic.ttf \
--- git/configure.ac.orig	2015-08-05 15:00:15.892758903 +0200
+++ git/configure.ac	2015-08-05 15:02:10.604798664 +0200
@@ -294,6 +294,8 @@
 lib/python/Plugins/Extensions/SocketMMI/Makefile
 lib/python/Plugins/Extensions/SocketMMI/meta/Makefile
 lib/python/Plugins/Extensions/SocketMMI/src/Makefile
+lib/python/Plugins/Extensions/TuxboxPlugins/Makefile
+lib/python/Plugins/Extensions/TuxboxPlugins/meta/Makefile
 lib/python/Plugins/PLi/Makefile
 lib/python/Plugins/SystemPlugins/DefaultServicesScanner/Makefile
 lib/python/Plugins/SystemPlugins/DefaultServicesScanner/meta/Makefile
@@ -332,7 +334,6 @@
 lib/python/Plugins/SystemPlugins/Videomode/meta/Makefile
 lib/python/Plugins/SystemPlugins/WirelessLan/Makefile
 lib/python/Plugins/SystemPlugins/WirelessLan/meta/Makefile
-lib/python/Plugins/Extensions/ModemSettings/Makefile
 lib/python/Plugins/SystemPlugins/OnErrorRestartService/Makefile
 lib/python/Tools/Makefile
 po/Makefile
--- git/lib/python/Plugins/Extensions/Makefile.am.orig	2015-08-05 15:01:25.736782905 +0200
+++ git/lib/python/Plugins/Extensions/Makefile.am	2015-08-05 15:03:29.700827016 +0200
@@ -1,10 +1,6 @@
 installdir = $(pkglibdir)/python/Plugins/Extensions
 
-SUBDIRS = CutListEditor DVDPlayer PicturePlayer MediaScanner MediaPlayer GraphMultiEPG SocketMMI
-
-if ! ENABLE_MINIMAL
-SUBDIRS += ModemSettings
-endif
+SUBDIRS = TuxboxPlugins CutListEditor DVDPlayer PicturePlayer MediaScanner MediaPlayer GraphMultiEPG SocketMMI
 
 install_PYTHON =	\
 	__init__.py
--- git/m4/ax_python_devel.m4.orig	2015-08-05 15:06:22.700891125 +0200
+++ git/m4/ax_python_devel.m4	2015-08-04 21:30:39.125636841 +0200
@@ -154,7 +154,7 @@
 		python_path=`$PYTHON -c "import distutils.sysconfig; \
 			print (distutils.sysconfig.get_python_inc ());"`
 		if test -n "${python_path}"; then
-			python_path="-I$PY_PATH/include/python$PYTHON_VERSION"
+			python_path="-I$python_path"
 		fi
 		PYTHON_CPPFLAGS=$python_path
 	fi
@@ -235,8 +235,7 @@
 			  "from distutils.sysconfig import get_python_lib as f; \
 			  import os; \
 			  print (os.path.join(f(plat_specific=1, standard_lib=1), 'config'));"`
-			PYTHON_LDFLAGS="-L$PY_PATH/include/python$PYTHON_VERSION -lpython$ac_python_version"
-#			PYTHON_LDFLAGS="-L$ac_python_libdir -lpython$ac_python_version"
+			PYTHON_LDFLAGS="-L$ac_python_libdir -lpython$ac_python_version"
 		fi
 
 		if test -z "PYTHON_LDFLAGS"; then
@@ -254,9 +253,8 @@
 	#
 	AC_MSG_CHECKING([for Python site-packages path])
 	if test -z "$PYTHON_SITE_PKG"; then
-		PYTHON_SITE_PKG="$PY_PATH/lib/python$PYTHON_VERSION/site-packages"
-#		PYTHON_SITE_PKG=`$PYTHON -c "import distutils.sysconfig; \
-#			print (distutils.sysconfig.get_python_lib(0,0));"`
+		PYTHON_SITE_PKG=`$PYTHON -c "import distutils.sysconfig; \
+			print (distutils.sysconfig.get_python_lib(0,0));"`
 	fi
 	AC_MSG_RESULT([$PYTHON_SITE_PKG])
 	AC_SUBST([PYTHON_SITE_PKG])
