--- a/Modules/_ctypes/_ctypes.c	2016-02-01 00:52:37.702465888 +0200
+++ b/Modules/_ctypes/_ctypes.c	2016-02-01 01:37:15.555744645 +0200
@@ -645,7 +645,13 @@
                      "symbol '%s' not found",
                      name);
 #else
-        PyErr_SetString(PyExc_ValueError, ctypes_dlerror());
+        const char *err = ctypes_dlerror();
+        if (!err)
+            PyErr_Format(PyExc_ValueError,
+                        "symbol '%s' not found",
+                         name);
+        else
+            PyErr_SetString(PyExc_ValueError, err);
 #endif
         return NULL;
     }
@@ -3396,7 +3402,13 @@
                      "function '%s' not found",
                      name);
 #else
-        PyErr_SetString(PyExc_AttributeError, ctypes_dlerror());
+        const char *err = ctypes_dlerror();
+        if (!err)
+            PyErr_Format(PyExc_AttributeError,
+                         "function '%s' not found",
+                         name);
+        else
+            PyErr_SetString(PyExc_AttributeError, err);
 #endif
         Py_DECREF(ftuple);
         return NULL;
